packages/mdc-segmented-button/segment/_segment-theme.scss

```diff
@@ -45,16 +44,21 @@ $border-radius: 4px;
 $icon-width: 24px;
 $icon-font-size: 18px;
 $label-padding: 6px;
+$touch-target-height: 48px;
 
 /// Sets the border color of the segment
 /// @param {String} $color - Color of segment outline
 @mixin outline-color($color, $query: feature-targeting.all()) {
+  $feat-color: feature-targeting.create-target($query, color);
+  $property: custom-properties.create(
+    --mdc-segmented-button-outline-color,
+    $color
+  );
+
   .mdc-segmented-button__segment {
-    $property: custom-properties.create(
-      --mdc-segmented-button-outline-color,
-      $color
-    );
-    @include _outline-color($property, $query);
+    @include feature-targeting.targets($feat-color) {
+      @include custom-properties.apply('border-color', $property);
```

Comment:
replace all `.apply()` call with `theme.property()`. I recently learnt that .apply() does not resolve tokens.